addons/account/tests/test_account_move_out_invoice.py

```diff
@@ -3590,3 +3592,21 @@ def test_invoice_mass_posting(self):
             self.assertEqual(aml.partner_id, self.partner_a)
         for aml in invoice2.line_ids:
             self.assertEqual(aml.partner_id, self.partner_b)
+
+    def test_journal_currency_change(self):
+        """
+        Test amount currency and balance are correctly recomputed when updating journal
+        and currency in the same write
+        """
+        invoice = self.init_invoice(move_type='out_invoice', partner=self.partner_a, invoice_date='2016-01-20', products=self.product_a, currency=self.currency_data['currency'])
+
+        journal_a = self.company_data['default_journal_sale']
+        journal_b = self.company_data['default_journal_sale'].copy()
+
+        currency_b = self.currency_data['currency']
+
+        for journal in (journal_b, journal_a):
+            with Form(invoice) as move_form:
+                move_form.journal_id = journal
+                move_form.currency_id = currency_b
+            self.assertEqual(invoice.amount_tax, 450.0, "Tax amount is not correct")
```

Comment:
Where does this amount come from?

Reply:
It's the initial invoice tax amount, at the end of the loop final value should match the initial value, I'll make it more clear